include/ignition/gazebo/comms/MsgManager.hh

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_MSGMANAGER_HH_
+#define IGNITION_GAZEBO_MSGMANAGER_HH_
+
+#include <deque>
+#include <map>
+#include <memory>
+#include <string>
+
+#include <ignition/transport/Node.hh>
+#include <ignition/utils/ImplPtr.hh>
+#include "ignition/gazebo/config.hh"
+#include "ignition/gazebo/System.hh"
+
+namespace ignition
+{
+namespace msgs
+{
+  // Forward declarations.
+  class Dataframe;
+}
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace comms
+{
+
+/// \brief A queue of message pointers.
+using DataQueue = std::deque<msgs::DataframeSharedPtr>;
+
+/// \brief A map where the key is the topic subscribed to an address and
+/// the value is a publisher to reach that topic.
+using SubscriptionHandler = std::map<std::string, transport::Node::Publisher>;
+
+/// \brief All the information associated to an address.
+struct AddressContent
+{
+  /// \brief Queue of inbound messages.
+  public: DataQueue inboundMsgs;
+
+  /// \brief Queue of outbound messages.
+  public: DataQueue outboundMsgs;
+
+  /// \brief Subscribers.
+  public: SubscriptionHandler subscriptions;
+
+  /// \brief Model name associated to this address.
+  public: std::string modelName;
```

Comment:
Should this be the Entity instead? Or will a fully scoped model name work better? I was wondering because when I was thinking of how to implement the acoustic comms subsystem as we would have to look up the entity pose and velocity.

Reply:
I thought about that as well. The only problem that I see is that the broker/MsgManager need to know about the ECM and so far they're very agnostic about all their job. If there's any clean way of doing it we could consider it. Otherwise we'd need to query the entity from the model in the derived comms sytem.